packages/services/src/assets/stacks-asset.utils.ts

```diff
@@ -0,0 +1,92 @@
+import { hexToCV } from '@stacks/transactions';
+
+import {
+  CryptoAssetCategories,
+  CryptoAssetChains,
+  CryptoAssetProtocols,
+  Sip9CryptoAssetInfo,
+  Sip10CryptoAssetInfo,
+} from '@leather.io/models';
+import { getTicker, isUndefined } from '@leather.io/utils';
+
+import {
+  HiroFtMetadataResponse,
+  HiroNftMetadataResponse,
+} from '../infrastructure/api/hiro/hiro-stacks-api.client';
+
+export function isTransferableSip10Token(token: Partial<HiroFtMetadataResponse>) {
+  return !isUndefined(token.decimals) && !isUndefined(token.name) && !isUndefined(token.symbol);
+}
+
+export function getAssetNameFromIdentifier(assetIdentifier: string) {
+  return !assetIdentifier.includes('::') ? assetIdentifier : assetIdentifier.split('::')[1];
+}
+
+export function getContractPrincipalFromAssetIdentifier(assetIdentifier: string) {
+  return assetIdentifier.split('::')[0];
+}
+
+export function getAddressFromAssetIdentifier(assetIdentifier: string) {
+  const principal = getContractPrincipalFromAssetIdentifier(assetIdentifier);
+  return principal.split('.')[0];
+}
```

Comment:
I recently saw this comment in extension

<img width="723" alt="image" src="https://github.com/user-attachments/assets/89a93cbb-da2b-420a-8b49-f40881634039" />

Logic being copied because of where its been placed. These fns are discreet pieces of logic useful to consume everywhere, without bringing with it a dependency on our entire services infra. cc @fbwoolf

Reply:
I think until we decide differently, utils like these should be located and made available from the Stacks pkg so they aren't duplicated. Rn it is tough bc the services pkg isn't available in the extension, correct?

Reply:
I agree chain specific package utils is the place for chain specific stuff like this. 

I think once things stabilise on mobile that the balances and activity service are prime candidates to integrate into the extension and before we know it we will be sharing loads of code with consistent numbers